[Reuben Chay] iP

ReubenChay opened the pull request:
> "Look at me I'm Mr Meseeks!" - Mr Meseeks 
# **Mr Meseeks**

Mr Meseeks is a revolutionary task manager that will improve your life! It's: 
- Text-Based using Command Line Interface (CLI) 
- User-Friendly 
- ~~Fast~~ SUPER FAST to use! 
- ~~S$1 ONLY~~ now **FREE** to use 👍 

Managing your tasks can be easy with **Mr Meseeks**, just follow these steps: 
1. download it [here](https://github.com/ReubenChay/ip/releases/tag/A-Release)
2. move the file to a folder (or anywhere which is convenient for you) 
3. open the file 
4. start using Mr Meseeks to manage your tasks for you! 😸 

Features of Mr Meseeks: 
- [x] Manage your tasks for you 
- [x] Manage Deadlines, To-Do tasks, Events 
- [x] User Interface  

Some examples of commands: 
``` 
list 
todo CS9999 Assignment 
delete 1 
mark 1 
```

If you are a Java programmer, you can use it to practice as well! Here's the main method: 
```ruby 
public static void main(String[] args) { 
        new Duke().run(); 
}
```

Commit:
Add Gradle support

Commit:
add Duke.java

Commit:
add Duke.java

Commit:
add Duke.java
add Task.java

Commit:
add Duke.java
add Task.java
add ToDo.java
add Event.java
add Deadline.java

Commit:
add Duke.java
add TaskStatusException.java
add DukeException.java
add IncompleteTaskNameException.java
add MissingDateException.java

Commit:
Duke.java

Commit:
add Duke.java

Commit:
modified:   Deadline.java
	modified:   Duke.java
	modified:   Event.java
	modified:   Task.java
	modified:   ToDo.java
	deleted:    ../../tasks.txt

Commit:
add Duke.java
add *.class

Commit:
Modified level-8, added javadoc

Commit:
Added Automatic Testing of Text UIs

Commit:
Modified for A-OOP

Commit:
Added Parser OOP

Commit:
Removed .class files, added packages

Commit:
Removed .class files, added packages

Commit:
Fixed some issues where Main driver could not be found

wrewsama commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,275 @@
-    public static void main(String[] args) {
+    public static void main(String[] args) throws DukeException, IllegalArgumentException {
+
+        List<Task> list = new ArrayList<>(); // to store list of inputs
```
I think the name of this list could be more descriptive

ReubenChay replied:
alright, thanks!

wrewsama commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,275 @@
+            try {
+                input = sc.nextLine();
+
+                if (input.equals("bye")) {
```
A switch case could be used here instead of if-elses

ReubenChay replied:
that makes a lot more sense, dont know why i didn't think of that. Thank you!

wrewsama commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,275 @@
+            for (String x : tasks) {
+                String taskType = x.substring(1,2);
+                Boolean done = x.substring(4,5).equals("X");
+                String[] arr = x.substring(6).split("\\(");
```
I think this name could be more descriptive

wrewsama reviewed (commented):
Some names could be more descriptive but overall LGTM!!

Commit:
Some fixes

Commit:
no message

Commit:
On branch Level-8
  Changes to be committed:
	modified:   src/main/java/Duke/Duke.java

Commit:
no message

Commit:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip into GradleWrapper

Pulling gradle support files

Commit:
Added Gradle Support

Commit:
Added JUnit Testing

Commit:
added checkstyle.xml files

Commit:
fixed slight error in file directory for checkstyle configs

Commit:
Added JavaDocs for some public classes and methods

Commit:
Made changes to abide by coding standard

Commit:
Added find function for Level-9

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Duke/TaskList.java

Commit:
Minor change to abide by coding standard

Commit:
Dummy commit to fix undetected A-Packages tag

Commit:
Cleaned up code. Properly used parser for more OOP

Commit:
Added checkstyle

Commit:
no message

Commit:
Fixed orphan commit of A-Packages

Commit:
Started on introducing javaFX

Commit:
More changes to add FXML

Commit:
Fixed some issues with FXML

Commit:
Added assertions

ReubenChay committed:
Merge pull request #2 from ReubenChay/branch-A-Assertions

Added assertions

Commit:
Update Duke.java
update Storage.java
update Parser.java

Method comments were a little confusing in Duke.java and Parser.java

This makes it difficult to read and understand their usages.

Improving comment readability allows others to understand the code
more easily and improve overall code quality.

Parser's parse method was long and complicated.

This makes it difficult to understand the logic of the code

Efforts were made to improve readability, but method remains a litte
long  as the current implementation requires a drastic change
in the base code of most other classes. Readability and code logic
was improved as much as possible.

ReubenChay committed:
Merge pull request #3 from ReubenChay/branch-A-CodeQuality

Merging PR from branch A-CodeQuality

Commit:
in the midst of building BCD extension C-update

Commit:
Updated Duke.java

When GUI was opened with FXML, initial items in file was not being updated.
When running in CLI, this problem did not exist. This was due to the fact
that storage.openFile() and storage.listInit() was only called in Duke.run()
method, which was not called when the GUI was run.

Methods mentioned were added to the constructor to ensure that the
tasks saved in the hard disk was properly added to the list when run
in the GUI.

Ensured that the program runs as intended both when using CLI and
when using the GUI.

Commit:
Updated Parser.java
Updated StringReplacer.java

Users did not have an option to edit their tasks without first
deleting them and re-adding them back.

An edit function was added to allow editing of both deadline and
event tasks (tasks with date and time). This allows users to
update the date and time of such events in the case of a change
in their schedule without having to first delete the task.

Update can be called with the following command:
update <task number> /<date and time>

Commit:
Merge branch 'branch-BCD'

Commit:
Updated MainWindow.java, Parser.java, StringReplacer.java, TaskList.java
Added jerry.png

GUI was not functioning properly in a few aspects. Some commands were
wrongly listed as IllegalArguments. This was due to trailing white spaces, which
were fixed with a simple String trim() method.

GUI would not close when 'bye' command was supplied. This was fixed
through adding a system.exit() signal in the MainWindow.java file. GUI closes
when intended now.

Bugs were fixed regarding the new BCD-Extension: Update tasks. Tasks
can now be updated bug-free.

In line with adding more personality, more catch phrases were added and
the GUI userImage was changed to fit the theme more (from TV show
Rick & Morty).

Commit:
Updated to keep CLI and GUI welcome message the same

Commit:
Added more personality

Commit:
Added code-reuse acknowledgement in MainWindow.java

Commit:
no message

Commit:
Added comment for class StringReplacer

Commit:
Merge branch 'branch-Level-10'

ReubenChay committed:
Added Ui.png 

A screenshot of the app

ReubenChay committed:
A-UserGuide

Added a user guide for the app

ReubenChay committed:
Update README.md

Commit:
no message

Commit:
Merge branch 'master' of https://github.com/ReubenChay/ip
